src/main/kotlin/org/jitsi/rtp/rtp/RtpPacket.kt

```diff
@@ -173,17 +174,62 @@ open class RtpPacket(
         get() = "type=RtpPacket len=$payloadLength " +
             "hashCode=${buffer.hashCodeOfSegment(payloadOffset, payloadOffset + payloadLength)}"
 
+    private var pendingHeaderExtensions: MutableList<HeaderExtension>? = null
+
+    private fun getEncodedHeaderExtension(extensionId: Int): HeaderExtension? {
+        if (!hasEncodedExtensions) return null
+
+        encodedHeaderExtensions.forEach { ext ->
+            if (ext.id == extensionId) {
+                return ext
+            }
+        }
+        return null
+    }
+
+    val hasExtensions: Boolean
+        get() = pendingHeaderExtensions?.isNotEmpty() ?: hasEncodedExtensions
```

Comment:
If `pendingHeaderExtensions` is empty (but not null), this will return `false` regardless of the value of `hasEncodedExtensions`. Is this the intent?